cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs

```diff
@@ -667,96 +667,164 @@ fn parse_url(
+                    .to_string(),
+                expected_registry_host_port_namespaced_string:
+                    "sat-r220-02.lab.eng.rdu2.redhat.com:5000/ns1".to_string(),
+            },
+            TestCase {
+                input: "quay.io".to_string(),
+                expected_registry: Registry {
                     scheme: "".to_string(),
                     insecure: false,
                     host: "quay.io".to_string(),
                     port: None,
                     namespace: "/".to_string(),
                 },
-            ),
-            (
-                "quay.io/ns1",
-                Registry {
+                expected_registry_host_port_string: "quay.io".to_string(),
+                expected_registry_host_port_namespaced_string: "quay.io/".to_string(),
+            },
+            TestCase {
+                input: "quay.io/ns1".to_string(),
+                expected_registry: Registry {
                     scheme: "".to_string(),
                     insecure: false,
                     host: "quay.io".to_string(),
                     port: None,
                     namespace: "/ns1".to_string(),
                 },
-            ),
-        ];
-
-        for (input, expected) in tests {
-            let registry: Registry = Registry::try_from_str(input)
-                .unwrap_or_else(|_| panic!("could not parse {} to registry", input));
-            assert_eq!(expected, registry);
-            assert_eq!(input, registry.host_port_string());
+                expected_registry_host_port_string: "quay.io".to_string(),
+                expected_registry_host_port_namespaced_string: "quay.io/ns1".to_string(),
+            },
+        ]
+    }
+
+    #[test]
+    fn registry_try_parse_valid() {
+        let tests = setup_test_cases();
+        for test in tests {
+            let registry: Registry = Registry::try_from_str(&test.input)
+                .unwrap_or_else(|_| panic!("could not parse {} to registry", &test.input));
+            assert_eq!(test.expected_registry, registry);
+        }
+    }
+
+    #[test]
+    fn registry_host_port_string() {
+        let tests = setup_test_cases();
+        for test in tests {
+            let registry: Registry = Registry::try_from_str(&test.input)
+                .unwrap_or_else(|_| panic!("could not parse {} to registry", &test.input));
+            assert_eq!(
+                test.expected_registry_host_port_string,
+                registry.host_port_string()
+            );
+        }
+    }
+
+    #[test]
+    fn registry_host_port_namespaced_string() {
+        let tests = setup_test_cases();
+        for test in tests {
+            let registry: Registry = Registry::try_from_str(&test.input)
+                .unwrap_or_else(|_| panic!("could not parse {} to registry", &test.input));
+            assert_eq!(
+                test.expected_registry_host_port_namespaced_string,
+                registry.host_port_namespaced_string()
+            );
```

Comment:
TBH I'm not sure whether this is worth a separate test, I'd just add that assertion to the previous one 

/shrug